File: server/src/protocol/BlindsUPBuilder.cpp
#include "BlindsUPBuilder.h"
#include "BlindsUPCommand.h"

#include <utility/BytesConverter.h>

using namespace std;
using namespace utility;
using namespace communication;

BlindsUPBuilder::BlindsUPBuilder()
{}

BlindsUPBuilder::~BlindsUPBuilder()
{}

unique_ptr<Command> BlindsUPBuilder::create(const vector<uint8_t> &commandInBytes)
{
    auto command = make_unique<BlindsUPCommand>();

    return move(command);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/BlindsUPBuilder.java
package wpam.mobile_client.protocol;
import java.io.Closeable;
import java.util.ArrayList;

public final class BlindsUPBuilder implements CommandBuilder, Closeable
	{
		public BlindsUPBuilder()
		{
		}
		public final void close()
		{
		}

		@Override
		public Command create(ArrayList<Integer> commandInBytes)
		{
			Command command  = new BlindsUPCommand();
			return command;
		}
	}